pkgs/shells/fish/default.nix

```diff
@@ -179,13 +179,8 @@ let
     postInstall = with lib; ''
       sed -r "s|command grep|command ${gnugrep}/bin/grep|" \
           -i "$out/share/fish/functions/grep.fish"
-      sed -i "s|which |${which}/bin/which |"               \
-             "$out/share/fish/functions/type.fish"
       sed -e "s|\|cut|\|${coreutils}/bin/cut|"             \
           -i "$out/share/fish/functions/fish_prompt.fish"
-      sed -e "s|gettext |${gettext}/bin/gettext |"         \
-          -e "s|which |${which}/bin/which |"               \
-          -i "$out/share/fish/functions/_.fish"
       sed -e "s|uname|${coreutils}/bin/uname|"             \
           -i "$out/share/fish/functions/__fish_pwd.fish"   \
```

Comment:
Not really related to your changes but.. I've been wondering why these are necessary. Are binaries like `uname` not in PATH?

Also some of these substitutions are likely out of date.
The "awk" substitution below is missing at least `share/functions/__fish_complete_wireshark.fish`.
You should probably add some checks that fail when new uses of sed and friends are added.

Also it seems strange that you don't change the path of coreutils' `seq` which is used quite a lot in completions.

Reply:
Thanks for your comment! I guess you're right for `coreutils` binaries as the package is part of the `propagatedBuildInputs`. On these grounds, patching `uname`, `seq`, etc. seems superfluous. Maybe @cole-h can shed some light on this.

Reply:
I'm just guessing, but I think referring to the absolute paths is more pure. If you built an environment with multiple packages depending on different versions of coreutils, they might pick up the wrong ones if they just pick up the first one that's available on the PATH. Setting the absolute paths here ensures they always refer to the build input coreutils at runtime.

Reply:
Basically what @r-burns said.

(Note that -- from what I understand -- `propagatedBuildInputs` does not affect the runtime PATH of any binary -- it's only for packages that include `fish` as a `buildInput`.)

I've had various TODOs regarding this whole `fish` situation on my list for months now, but haven't had the time (or, more accurately, the mental bandwidth) to dig into it.

Some slightly related discussion happened here: https://github.com/NixOS/nixpkgs/pull/92229.

Reply:
> (Note that -- from what I understand -- propagatedBuildInputs does not affect the runtime PATH of any binary; it's only for packages that include fish as a buildInput.)

Yes, you're right – my bad. So, the answer to your question, @krobelus, is that those binaries are _not necesarily_ in `PATH`. It is, however, not unlikely that we forgot to patch some commands. Probably just nobody noticed as users mostly have packages like `coreutils` in their `PATH`. 

I just noted that we don't run the tests. I guess, we should change that unless there's a specific reason that we don't.

Reply:
I added a commit to run the tests. Currently, they still run with `coreutils` as `checkInputs` which is probably not what we want.